sources/api/apiserver/src/server/controller.rs

```diff
@@ -420,21 +497,92 @@ pub(crate) fn get_metadata_for_data_keys<D: DataStore, S: AsRef<str>>(
     Ok(result)
 }
 
-/// Gets the value of a metadata key everywhere it's found in the data store.  Returns a mapping
-/// of data key to the metadata value associated with the requested key.
-pub(crate) fn get_metadata_for_all_data_keys<D: DataStore, S: AsRef<str>>(
+// Sort the given settings hashmap where a parent
+// key is always before its successors.
+fn sort_metadata(metadata: HashMap<Key, HashMap<Key, String>>) -> Vec<(Key, HashMap<Key, String>)> {
+    let mut metadata_sorted: Vec<_> = metadata.into_iter().collect();
+    metadata_sorted.sort_by(|(k1, _), (k2, _)| k1.segments().len().cmp(&k2.segments().len()));
+    metadata_sorted.to_vec()
+}
+
+/// Settings generators can be dynamically applied to a set of settings.
+///
+/// This function resolves a dynamically expanded generator into a finite set of
+/// distinct settings generators.
+fn expand_setting_generator<D: DataStore>(
+    datastore: &D,
+    data_key: &Key,
+    raw_setting_generator: RawSettingsGenerator,
+    result: &mut HashMap<String, Value>,
+) -> Result<()> {
+    let depth = raw_setting_generator.depth;
+    let metadata_value = serde_json::to_value(SettingsGenerator::from(raw_setting_generator))
+        .context(error::SerializeSnafu)?;
+
+    if depth == 0 {
+        // If the depth is 0, we don't need to process the successors
+        result.insert(data_key.to_string(), metadata_value);
+        return Ok(());
+    } else {
```

Comment:
You can remove this and out-dent the remainder of the function body, since it'll always run, and won't run in the `depth == 0` case now because of the early return.